homeassistant/components/teslemetry/__init__.py

```diff
@@ -69,8 +72,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: TeslemetryConfigEntry) -
         access_token=access_token,
     )
     try:
-        scopes = (await teslemetry.metadata())["scopes"]
-        products = (await teslemetry.products())["response"]
+        calls = await asyncio.gather(
+            teslemetry.metadata(),
+            teslemetry.products(),
+        )
+        scopes = calls[0]["scopes"]
+        region = calls[0]["region"]
+        products = calls[1]["response"]
```

Comment:
Can this raise? Otherwise, move it out of the try block

Reply:
Done